pkg/ingester/ingester_v2.go

```diff
@@ -2245,3 +2282,11 @@ func (i *Ingester) getInstanceLimits() *InstanceLimits {
 
 	return l
 }
+
+func (i *Ingester) checkIfAllTSDBClosing() bool {
```

Comment:
Thank you for the suggestion. If it is using block store, it will check if the state is running. Otherwise it will check if it is running or stopping. If the ingester is stopping, it will return an error before it even gets to the v2 functions. 

If I misunderstood what you had in mind, please let me know and I will try to update it again.

Reply:
I think Peter was suggesting the code look like:

```Go
func (i *Ingester) Query(ctx context.Context, req *client.QueryRequest) (*client.QueryResponse, error) {
	if i.cfg.BlocksStorageEnabled {
		return i.v2Query(ctx, req)
	}

	if err := i.checkRunningOrStopping(); err != nil {
		return nil, err
	}
...
```

```Go
func (i *Ingester) v2Query(ctx context.Context, req *client.QueryRequest) (*client.QueryResponse, error) {
	if err := i.checkRunning(); err != nil {
		return nil, err
	}
...
```

Reply:
Oh I see now. I have updated it. Thanks for the help!